File: src/main/java/gargoyle/ct/messages/LocaleProvider.java
package gargoyle.ct.messages;

import gargoyle.ct.pref.CTPreferences.SUPPORTED_LOCALES;
import gargoyle.ct.prop.CTObservableProperty;

import java.util.Locale;

public interface LocaleProvider {
    Locale getLocale();

    void setLocale(Locale locale);

    CTObservableProperty<SUPPORTED_LOCALES> locale();
}

File: src/main/kotlin/gargoyle/ct/messages/LocaleProvider.kt
package gargoyle.ct.messages

import gargoyle.ct.pref.CTPreferences.SupportedLocales
import gargoyle.ct.prop.CTObservableProperty
import java.util.Locale

interface LocaleProvider {
    fun getLocale(): Locale
    fun setLocale(locale: Locale)
    fun locale(): CTObservableProperty<SupportedLocales>
}
